cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -507,14 +507,16 @@ void reader::impl::create_global_chunk_info()
   }
 }
 
-void reader::impl::compute_input_passes()
+void reader::impl::compute_input_passes(read_mode mode)
 {
   // at this point, row_groups has already been filtered down to just the row groups we need to
   // handle optional skip_rows/num_rows parameters.
   auto const& row_groups_info = _file_itm_data.row_groups;
 
-  // if the user hasn't specified an input size limit, read everything in a single pass.
-  if (_input_pass_read_limit == 0) {
+  // If we are reading all rows at once, read everything in a single pass. We can't use
+  // `_input_pass_read_limit` as test here for `CHUNKED_READ` mode as we may need to create a pass
+  // using max column size limits
+  if (mode == read_mode::READ_ALL) {
```

Comment:
Don't rely on the `_input_pass_read_limit` here as we may be reading `>2B` rows with no limits with the chunked reader.